queries/reflect.go

```diff
@@ -301,24 +302,32 @@ func PtrsFromMapping(val reflect.Value, mapping []uint64) []interface{} {
 func ValuesFromMapping(val reflect.Value, mapping []uint64) []interface{} {
 	ptrs := make([]interface{}, len(mapping))
 	for i, m := range mapping {
-		ptrs[i] = ptrFromMapping(val, m, false).Interface()
+		ptrs[i] = ptrFromMapping(val, m, false)
 	}
 	return ptrs
 }
 
 // ptrFromMapping expects to be passed an addressable struct that it's looking
 // for things on.
-func ptrFromMapping(val reflect.Value, mapping uint64, addressOf bool) reflect.Value {
+func ptrFromMapping(val reflect.Value, mapping uint64, addressOf bool) interface{} {
+
+	if mapping == uint64(0) {
+		var ignored interface{}
+		return &ignored
```

Comment:
This causes an allocation on the heap every time because `ignored` will escape. Though it could be okay because this is in an exceptional case.

Reply:
I agree with you, I think it could be ok as an exceptional case.